openpype/hosts/maya/plugins/publish/extract_pointcache.py

```diff
@@ -52,41 +52,48 @@ def process(self, instance):
         filename = "{name}.abc".format(**instance.data)
         path = os.path.join(parent_dir, filename)
 
-        options = {
-            "step": instance.data.get("step", 1.0),
-            "attr": attrs,
-            "attrPrefix": attr_prefixes,
-            "writeVisibility": True,
-            "writeCreases": True,
-            "writeColorSets": instance.data.get("writeColorSets", False),
-            "writeFaceSets": instance.data.get("writeFaceSets", False),
-            "uvWrite": True,
-            "selection": True,
-            "worldSpace": instance.data.get("worldSpace", True)
-        }
+        options = {"selection": True}
+        option_keys = [
+            "dataFormat",
+            "eulerFilter",
+            "noNormals",
+            "preRoll",
+            "preRollStartFrame",
+            "renderableOnly",
+            "step",
+            "stripNamespaces",
+            "uvWrite",
+            "verbose",
+            "wholeFrameGeo",
+            "worldSpace",
+            "writeColorSets",
+            "writeCreases",
+            "writeFaceSets",
+            "writeUVSets",
+            "writeVisibility",
+        ]
+        for key in option_keys:
+            if key in instance.data:
+                options[key] = instance.data[key]
 
         if not instance.data.get("includeParentHierarchy", True):
             # Set the root nodes if we don't want to include parents
             # The roots are to be considered the ones that are the actual
             # direct members of the set
             options["root"] = roots
 
-        if int(cmds.about(version=True)) >= 2017:
-            # Since Maya 2017 alembic supports multiple uv sets - write them.
-            options["writeUVSets"] = True
-
         if instance.data.get("visibleOnly", False):
             # If we only want to include nodes that are visible in the frame
             # range then we need to do our own check. Alembic's `visibleOnly`
             # flag does not filter out those that are only hidden on some
             # frames as it counts "animated" or "connected" visibilities as
             # if it's always visible.
-            nodes = list(iter_visible_nodes_in_range(nodes,
-                                                     start=start,
-                                                     end=end))
+            nodes = list(
+                iter_visible_nodes_in_range(nodes, start=start, end=end)
+            )
```

Comment:
Due to now how the attributes can be "enabled" and "disabled" to expose to artist or not means we've now lost the default attribute values for `writeUVSets=True`, `writeCreases=True`, `writeVisibility=True`, `uvWrite=True`.

We should find a way where the settings can state what the default value should be and if not enabled/exposed to the artist it should _still_ use that default value.

Reply:
Does it visually make sense in the settings to have the "enabled" toggle and then the "default" value which will be used no matter what, like this (current state of PR)?

![Capture](https://github.com/ynput/OpenPype/assets/1860085/8e80a214-9fd1-462a-9ca8-f20f2997212d)

Reply:
It would if the Enabled label was "Artist customizable" or "Exposed to artist" or "Allow instance customization" or alike? With just "enabled" yes it'd be confusing.

Anyway, without getting at least the previous defaults back we're basically breaking cases where if we're disabling e.g. writeCreases as customizable then it will never write creases which seems off.

Reply:
We can fix the PR to make it backward compatible, but I need to know what I'm working with from the settings first hence why I'm trying to get the settings right first.

How about this?
![Capture](https://github.com/ynput/OpenPype/assets/1860085/b4c43cae-55df-42bd-9772-e5559da64138)

Reply:
That checkbox is intended to be the actual value, right? Not something that enables the default value? (Just checking since this just happens to be a boolean attribute.) If it's the actual value, that's clear to me.

Then another checkbox for each attribute saying "Allow instance customization" or whatever then that's what exposes it to artists.

Reply:
![Capture](https://github.com/ynput/OpenPype/assets/1860085/28a6d350-0644-4128-9885-0b26051a639e)

1. Exposes/displays the attribute to the artist.
2. The default value used when creating/publishing.

Reply:
I wouldn't have expected **1)** to be the expose to artist toggle somehow. But maybe I misread (or didnt read at all) that the label said "Display X attribute" to begin with.
Let's just try and begin with a design like that and go from there?

Reply:
I gotta say that wrapping this as categories is visually quite an overkill I think we could simplify it quite a bit by simply exposing for each attribute with two values. `Artist editable` and `default value`

In principle exactly what Toke has, I'm just wondering about the structure. @tokejepsen apply the same settings to ayon-addons settings please so we can see how it behaves there.

Reply:
@mkolar is this what you mean?

![Capture](https://github.com/ynput/OpenPype/assets/1860085/86969680-d213-490b-b229-8b9cd3c77d07)

Can have a look at AYON settings afterwards.

Reply:
Can we avoid the collapsible headers to make it less bulky?

- write uv default
- write uv artist editable
- other attr default
- other attr artist editable

Etc.

Reply:
Hows this?

![Capture](https://github.com/ynput/OpenPype/assets/1860085/0e25dea4-b0ee-423d-92e9-7000055ac7b0)

Reply:
@BigRoy does the above work for you?

Reply:
It's not as clear as I had hoped but I feel like most likely it's the best case where you still have a decent oversight over what's going on for many attributes.